File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.Models
{
    public class Transaction
    {
        [Key]
        public int TransactionId {  get; set; }
        [ForeignKey("Account")]
        public int AccountId {  get; set; }
        public virtual Account? Account { get; set; }
        [ForeignKey("Payment")]
        public int PayementId {  get; set; }
        public virtual Payment? Payment { get; set; }
        [Required(ErrorMessage ="Transaction Type is Required!")]
        [ForeignKey("TransactionType")]
        public int TransactionTypeId { get; set; }
        public virtual TransactionType? TransactionType { get; set; } 
        [Required(ErrorMessage = "Transaction Amount is Required!")]
        [DataType(DataType.Currency)]
        public int Amount {  get; set; }
        [Required(ErrorMessage = "Transaction Date is Required!")]
        [DataType(DataType.DateTime)]
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Models/Transaction.ts
import { Account } from "./Account";
import { Payment } from "./Payment";
import { SalaryDisbursement } from "./SalaryDisbursement";
import { TransactionType } from "./TransactionType";

export interface Transaction {
  transactionId: number;
  accountId: number;
  account?: Account;
  paymentId?: number;
  payment?: Payment;
  salaryDisbursementId?: number;
  salaryDisbursement?: SalaryDisbursement;
  toFrom:string;
  transactionTypeId: number;
  transactionType?: TransactionType;
  amount: number;
  createdAt: string; 
}